File: app/src/main/java/net/xzos/UpgradeAll/ui/viewmodels/log/PageViewModel.java
package net.xzos.UpgradeAll.ui.viewmodels.log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

import net.xzos.UpgradeAll.application.MyApplication;
import net.xzos.UpgradeAll.server.log.LogDataProxy;

import java.util.List;

public class PageViewModel extends ViewModel {

    private MutableLiveData<String[]> mLogObjectTag = new MutableLiveData<>();
    private LiveData<LiveData<List<String>>> mLogList = Transformations.map(mLogObjectTag, logObjectTag -> new LogDataProxy(MyApplication.getServerContainer().getLog()).getLogMessageListLiveData(logObjectTag));

    void setLogObjectTag(String[] logObjectTag) {
        mLogObjectTag.setValue(logObjectTag);
    }

    LiveData<LiveData<List<String>>> getLogList() {
        return mLogList;
    }
}

File: app/src/main/kotlin/net/xzos/upgradeAll/ui/viewmodels/log/PageViewModel.kt
package net.xzos.upgradeAll.ui.viewmodels.log

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel

import net.xzos.upgradeAll.server.ServerContainer
import net.xzos.upgradeAll.server.log.LogDataProxy

class PageViewModel : ViewModel() {

    private val mLogObjectTag = MutableLiveData<Array<String>>()
    internal val logList = Transformations.map(mLogObjectTag) { logObjectTag -> LogDataProxy(ServerContainer.Log).getLogMessageListLiveData(logObjectTag) }

    internal fun setLogObjectTag(logObjectTag: Array<String>) {
        mLogObjectTag.value = logObjectTag
    }
}